uint32 height
uint32 width
uint32 channel
# data
int8[] serialize_image